client/extensions/woocommerce/state/sites/settings/email/selectors.js

```diff
@@ -0,0 +1,153 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+// path to mailchim setting state branch
+const basePath = ( siteId ) => (
+	[
+		'extensions',
+		'woocommerce',
+		'sites',
+		siteId,
+		'settings',
+		'email',
+	]
+);
+
+/**
+ * Returns true if currently requesting mailchim settings or false otherwise.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @return {Boolean}        Whether settings are being requested
+ */
+export const isRequestingSettings = ( state, siteId ) => {
+	const path = [ ...basePath( siteId ), 'settingsRequest' ];
+
+	return get( state, path, false );
+};
+
+/**
+ * Returns settings request error object
+ *
+ * @param  {Object}  state   Global state tree
+ * @param  {Number}  siteId  Site ID
+ * @return {Object}          error object if present or null otherwise
+ */
+export const requestingSettingsError = ( state, siteId ) => {
+	const path = [ ...basePath( siteId ), 'settingsRequestError' ];
+
+	return get( state, path, null );
+};
+
+export const mailchimpSettings = ( state, siteId ) => {
+	const path = [ ...basePath( siteId ), 'settings' ];
+
+	return get( state, path, null );
+};
+
+/**
+ * Returns true if currently subbmitting MailChimp API key or false otherwise.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @return {Boolean}        Whether api key is being subbmitted
+ */
+export const isSubbmittingApiKey = ( state, siteId ) => {
+	const path = [ ...basePath( siteId ), 'apiKeySubmit' ];
+
+	return get( state, path, false );
+};
+
+/**
+ * Returns true if currently subbmitting MailChimp newsletter settings or false otherwise.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @return {Boolean}        Whether newsletter settings are being subbmitted
+ */
+export const isSubmittingNewsletterSetting = ( state, siteId ) => {
+	const path = [ ...basePath( siteId ), 'newsletterSettingsSubmit' ];
+
+	return get( state, path, false );
+};
+
+/**
+ * Returns newletter settings submit error object
+ *
+ * @param  {Object}  state   Global state tree
+ * @param  {Number}  siteId  Site ID
+ * @return {Object}          error object if present or null otherwise
+ */
+export const newsletterSettingsSubmitError = ( state, siteId ) => {
+	const path = [ ...basePath( siteId ), 'newsletterSettingsSubmitError' ];
+
+	return get( state, path, false );
+};
+
+/**
+ * Returns true if current api key is correct.
```

Comment:
Is it that it is correct, or that it is not empty?

Reply:
This is correct. The default true is because we assume api key to be true until we verify it.